Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/ClusterManagerEx.java

```diff
@@ -69,6 +74,54 @@ public static ClusterManagerEx getInstance() {
         return instance;
     }
 
+    public String getClusterConnectionString(final @NotNull String clusterName) {
+        if (isAzureChinaEnvironment()) {
+            return String.format("https://%s.azurehdinsight.cn", clusterName);
+        } else {
+            return String.format("https://%s.azurehdinsight.net", clusterName);
+        }
+    }
+
+    public String getBlobFullName(@NotNull final String storageName) {
+        return storageName + getBlobSuffix();
+    }
+
+    public String getBlobSuffix() {
+        if (isAzureChinaEnvironment()){
+            return ".blob.core.chinacloudapi.cn";
+        } else {
+            return "blob.core.windows.net";
+        }
+    }
+
+    public String getPortalUrl() {
+        if (isAzureChinaEnvironment()) {
+            return "https://portal.azure.cn/";
+        } else {
+            return "https://portal.azure.com/";
+        }
+    }
+
+    public Environment getAzureEnvironment() {
+        AzureManager azureManager = null;
```

Comment:
Suggest to use `Optional<>` for nullable reference.

Reply:
it's just for internal function var. I think we can just keep it.